src/provider/mod.rs

```diff
@@ -51,7 +52,36 @@ pub trait Provider {
     ///
     /// It must verify that each referenced parcel is present in storage. Any parcel that is not
     /// present must be returned in the list of labels.
-    async fn create_invoice(&self, inv: &super::Invoice) -> Result<Vec<super::Label>>;
+    async fn create_invoice(
+        &self,
+        inv: &mut super::Invoice, // Signing requires mutability
+        signing_role: SignatureRole,
+        secret_key: &SecretKeyEntry,
+        verification_strategy: VerificationStrategy,
+    ) -> Result<Vec<super::Label>>;
+
+    /// This is the default implementation of the signing logic. All providers should call
+    /// this function to sign the invoice when creating a new invoice.
+    fn sign_invoice(
+        &self,
+        inv: &mut super::Invoice,
+        role: SignatureRole,
+        secret_key: &SecretKeyEntry,
+    ) -> Result<()> {
+        inv.sign(role, secret_key)?;
+        Ok(())
+    }
+
+    // TODO: This can probably be removed.
```

Comment:
Any clarification on this comment? Are providers supposed to require verification when implementing?

Reply:
Removed.